### PR TITLE
linter: Enable error checking linter

### PR DESCRIPTION
**Pull request by shazbert**
# PR Description

@gloriousCode with his keen senses 🦅 🦅 🦅 🦅 🦅 🦅 🦅  found an error that wasn't being checked. I thought it prudent to adjust golangci-lint to enable error return checking to reduce missing them in the future. 

FTX: Fix bug where the asset was being enabled when already enabled which introduced a false positive error. 
Bitmex: Fix bug where an unsupported asset type was in the configuration and was returning a false positive error. Update reduces work. 

Fixes # (issue)

## Type of change

Please delete options that are not relevant and add an `x` in `[]` as item is complete.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How has this been tested

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration and
also consider improving test coverage whilst working on a certain feature or package.

- [ ] go test ./... -race
- [ ] golangci-lint run
- [x] Test X

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation and regenerated documentation via the documentation tool
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally and on Github Actions/AppVeyor with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: gctscript/vm/vm_test.go [hunk 1]
```diff
@@ -25,7 +25,6 @@ var (
 	testBrokenScript         = filepath.Join("..", "..", "testdata", "gctscript", "broken.gct")
 	testScriptRunner         = filepath.Join("..", "..", "testdata", "gctscript", "timer.gct")
 	testScriptRunner1s       = filepath.Join("..", "..", "testdata", "gctscript", "1s_timer.gct")
-	testScriptRunnerInvalid  = filepath.Join("..", "..", "testdata", "gctscript", "invalid_timer.gct")
```
Comment (gloriousCode) [inline gctscript/vm/vm_test.go:28]: Why was this removed?

File: go.sum [hunk 2]
```diff
@@ -451,6 +451,7 @@ golang.org/x/mod v0.2.0/go.mod h1:s0Qsj1ACt9ePp/hMypM3fl4fZqREWJwdYDEqhRiZZUA=
 golang.org/x/mod v0.3.0/go.mod h1:s0Qsj1ACt9ePp/hMypM3fl4fZqREWJwdYDEqhRiZZUA=
```
Comment (gloriousCode) [inline go.sum:451]: Can `go mod tidy`

**Review by gloriousCode — CHANGES_REQUESTED** [review]
Nice work Shazbert!

File: gctscript/vm/vm_test.go (hunk 1 above)
Comment (shazbert) [reply gctscript/vm/vm_test.go:28]: It was not being used.

Comment (gloriousCode) [reply gctscript/vm/vm_test.go:28]: The file is still there. I think we should re-add it, or remove the file if its a broken test

**Review by gloriousCode — APPROVED** [review]
tACK!
# 🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉

**Review by thrasher- — APPROVED** [review]
tACK, thanks also for cleaning up a lot of deadcode along the way!

File: exchanges/exchange_test.go [hunk 3]
```diff
@@ -1298,7 +1319,6 @@ func TestSetupDefaults(t *testing.T) {
 	// Test websocket support
 	b.Websocket = stream.New()
 	b.Features.Supports.Websocket = true
-	b.SetupDefaults(&cfg)
```
Comment (gbjk) [inline exchanges/exchange_test.go:1301]: Removing this makes this section of the test a duplicate of `TestIsWebsocketEnabled` :)
Will remove it.
